cmd/prometheus/main.go

```diff
@@ -181,6 +181,9 @@ func (c *flagConfig) setFeatureListOptions(logger log.Logger) error {
 			case "extra-scrape-metrics":
 				c.scrape.ExtraMetrics = true
 				level.Info(logger).Log("msg", "Experimental additional scrape metrics enabled")
+			case "metadata-storage":
+				c.scrape.EnableMetadataStorage = true
+				level.Info(logger).Log("msg", "Experimental in-memory metadata storage enabled")
```

Comment:
we should probably check that this is enabled when `SendWALMetadata` is being used, and if not print an error/warning

Reply:
Done in 42fc28821aafa9f42e6672343e748195ba83aa86, after parsing the configuration file. Not sure if there's a better place for the log line that has access to both fields.